src/ExpertExtender/ExpertExtender.CalendarHint.js

```diff
@@ -94,15 +94,17 @@
 				return;
 			}
 
-			// Are we in the interesting range
-			if( !( value.year() > 1581 && value.year() < 1930 && value.precision() > 10 ) ) {
+			var assumeCalendar = value.getYear() <= 1581 || value.getYear() >= 1930;
+
+			if( assumeCalendar || value.getOption( 'precision' ) <= 10 ) {
 				this.$calendarhint.hide();
 				return;
 			}
 
-			var msg = this._messageProvider.getMessage(
-				this._prefix + '-' + value.calendar().toLowerCase()
-			);
+			var calendarModel = value.getOption( 'calendarModel' ),
```

Comment:
Oh, and no, all I can do at the moment is looking at the code. I can not test anything here because of an "Error: Unknown dependency: mw.ext.dataValues" I do not understand.